packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -10,16 +10,33 @@ import 'package:flutter/painting.dart';
 import 'package:vector_math/vector_math_64.dart';
 
 import 'debug.dart';
+import 'node.dart';
 
 /// A composited layer.
 ///
 /// During painting, the render tree generates a tree of composited layers that
 /// are uploaded into the engine and displayed by the compositor. This class is
 /// the base class for all composited layers.
-abstract class Layer extends Object with TreeDiagnosticsMixin {
-  /// This layer's parent in the layer tree
-  ContainerLayer get parent => _parent;
-  ContainerLayer _parent;
+///
+/// Most layers can have their properties mutated. The layer tree as a whole can
+/// be reshaped also, with children being moved around the tree. The scene must
+/// be explicitly recomposited after such changes are made; the layer tree does
+/// not maintain its own dirty state. To composite the tree, create a
+/// [ui.SceneBuilder] object, pass it to the root [Layer] object's [addToScene]
+/// method, and then call [ui.SceneBuilder.build] to obtain a [Scene]. A [Scene]
+/// can then be painted using [ui.window.render].
+///
+/// See also:
+///
+///  * [RenderView.compositeFrame], which implements this recomposition protocol
+///    for painting [RenderObject] trees on the the display.
+abstract class Layer extends AbstractNode with TreeDiagnosticsMixin {
```

Comment:
I'm not sure it really matters. This is really low-level stuff in the framework. Unless there's a performance reason to do it (and I wouldn't be surprised if it was the opposite), I'd rather keep this simple.